### PR TITLE
TCF modal display data

### PR DESCRIPTION
**Pull request by allisonking**
Closes https://github.com/ethyca/fides/issues/3809

### Description Of Changes

https://www.loom.com/share/0c8b1c05f6be4f99aab109435d808a8a?sid=ebaa2503-68b3-43ff-aa07-b925982eddb5

#### Important caveats
* This work does not yet touch the `fides_consent` cookie. We likely want to initialize values in the cookie the same way we do for notices. We haven't talked too much about what the `notice_key`s for these will be yet though. https://github.com/ethyca/fides/issues/3886
* This also does not integrate with the new consent reporting flow. That should be done in https://github.com/ethyca/fides/issues/3877

### Code Changes

* [x] Display purposes, features, and vendors in tabs
* [x] Hook up toggles

### Steps to Confirm

* Run nox -s dev in fides to start the backend
* `tcf_enabled` needs to be set to True: make a PATCH to `/consent-settings` and set `tcf_enabled: true` in the request body
* Use a VPN to go to an EU location, i.e. France
* To add purposes, you need a system with at least one data use that belongs to a GVL purpose or special purpose.  For example, [data_use: "essential.fraud_detection" maps to Special Purpose #1](https://github.com/ethyca/fideslang/compare/main...asachs/tcf-mappings#diff-a4adb03db889d2228be2f2adbcfe3ee6a2964e133c37f190bc34ae8cf8b0a7fbR117)
* You can do this through the admin-ui: run `turbo dev` in `clients/admin-ui` and then visit localhost:3000
* In `clients/privacy-center`, run `turbo dev`. If you have the admin-ui running, this will start the privacy center at localhost:3001
* Visit localhost:3001/fides-js-demo.html
* The banner should appear and you should be able to click “Manage preferences” to see the TCF modal
* To add vendors, attach an integration to a system that has a GVL data use (i.e. set up an amplitude integration with the system from step 5)

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* [x] Issue Requirements are Met
* [x] Relevant Follow-Up Issues Created
  * [x] https://github.com/ethyca/fides/issues/3885
* [x] Update `CHANGELOG.md`

### CODE REVIEW

File: clients/fides-js/src/components/ConsentButtons.tsx [hunk 1]
```diff
@@ -122,41 +122,3 @@ export const NoticeConsentButtons = ({
     />
   );
 };
-
-interface TcfConsentButtonProps {
```
Comment (allisonking) [inline clients/fides-js/src/components/ConsentButtons.tsx:126]: moved this to its own file

File: clients/fides-js/src/components/Toggle.tsx [hunk 2]
```diff
@@ -28,7 +28,7 @@ const Toggle = ({
         onChange={() => {
           onChange(id);
         }}
-        defaultChecked={checked}
+        checked={checked}
```
Comment (allisonking) [inline clients/fides-js/src/components/Toggle.tsx:31]: this is kind of a weird thing, but it turns out `defaultChecked` only provides the initial value. this started not to work when I added the "select all/deselect toggles" which wanted to control _other_ toggles' state. This seems to work well, though I had to update the cypress tests now that the `checked` prop is a lil different

more info: https://stackoverflow.com/questions/33131413/react-defaultchecked-dont-render-check-attribute-on-second-load

File: clients/fides-js/src/components/tcf/TcfOverlay.tsx [hunk 3]
```diff
@@ -2,47 +2,110 @@ import { h, FunctionComponent } from "preact";
...
+import { TcfConsentButtons } from "./TcfConsentButtons";
 import { OverlayProps } from "../types";
 import TcfTabs from "./TcfTabs";
+import {
+  TCFFeatureRecord,
+  TCFPurposeRecord,
+  TCFVendorRecord,
+} from "../../lib/tcf/types";
+
+const resolveConsentValueFromTcfModel = (
```
Comment (allisonking) [inline clients/fides-js/src/components/tcf/TcfOverlay.tsx:22]: this is the kind of thing that can be taken out of this component and put in `lib` instead if we need to initialize the cookie value before rendering the overlay https://github.com/ethyca/fides/issues/3886

File: clients/fides-js/src/components/tcf/TcfOverlay.tsx [hunk 4]
```diff
@@ -2,47 +2,110 @@ import { h, FunctionComponent } from "preact";
...
+  model: TCFPurposeRecord | TCFFeatureRecord | TCFVendorRecord
+) => {
+  if (model.current_preference) {
```
Comment (allisonking) [inline clients/fides-js/src/components/tcf/TcfOverlay.tsx:25]: we don't handle notice-only yet—I originally didn't think there were "notice-only" TCF purposes, but I think there might be?

File: clients/fides-js/src/components/tcf/TcfVendors.tsx [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  onChange,
+}: {
+  allVendors: PrivacyExperience["tcf_vendors"];
+  enabledIds: string[];
+  onChange: (payload: UpdateEnabledIds) => void;
+}) => {
+  const [isFiltered, setIsFiltered] = useState(false);
+
+  if (!allVendors || allVendors.length === 0) {
+    // TODO: empty state?
```
Comment (eastandwestwind) [inline clients/fides-js/src/components/tcf/TcfVendors.tsx:46]: do we need product input on this?

File: clients/fides-js/src/components/tcf/TcfOverlay.tsx (hunk 4 above)
Comment (eastandwestwind) [reply clients/fides-js/src/components/tcf/TcfOverlay.tsx:25]: I'm unsure. @rsilvery - are you aware of any TCF purposes being notice-only?

**Pull request merged**